src/python/WMCore/Services/StompAMQ/StompAMQ.py

```diff
@@ -189,3 +189,51 @@ def make_notification(self, payload, docType, docId, producer=None, ts=None):
             docs.append(notification)
 
         return docs
+
+    def make_notification(self, payload, docType, docId, producer=None, ts=None, metadata=None,
+                          payload_subfield=False):
+        """
+        Produce a notification from a single payload, adding the necessary
+        headers and metadata. Generic metadata is generated to include a
+        timestamp, producer name, document id, and a unique id. User can
+        pass additional metadata which updates the generic metadata.
+
+        If payload already contains a metadata field, it is overwritten.
+
+        :param payload: Actual data.
+        :param docType: document type for metadata.
+        :param docId: document id representing the notification.
+        :param producer: The notification producer name.
+        :param ts: timestamp to be added to metadata.
+        :param metadata: user metadata to be added. (Updates generic metadata.)
+        :param payload_subfield: Toggle whether to put the payload in a subfield
+               or not
+
+        :return: a list of notifications with the proper metadata
+        """
+        producer = producer or self._producer
+        umetadata = metadata or {}
+        ts = ts or int(time.time())
+
+        headers = {'type': docType,
+                   'version': self._version,
+                   'producer': producer}
+
+        metadata = {'timestamp': ts,
+                    'producer': producer,
+                    'id': docId,
+                    'uuid': makeUUID()}
```

Comment:
given that `id` is already meant to be unique, I wonder whether we actually have any need for `uuid`?

Reply:
It doesn't make much difference for me either. @vkuznet @leggerf shall we keep uuid?